esmvalcore/preprocessor/_derive/sispeed.py

```diff
@@ -39,43 +39,13 @@ def calculate(cubes):
         """
         siu = cubes.extract_strict(Constraint(name='sea_ice_x_velocity'))
         siv = cubes.extract_strict(Constraint(name='sea_ice_y_velocity'))
-
         try:
             return DerivedVariable._get_speed(siu, siv)
         except ValueError:
-            # Models usually store siu and siv in slightly different points
-            if not DerivedVariable._coordinate_close(siu, siv, 'latitude'):
-                raise
-            if not DerivedVariable._coordinate_close(siu, siv, 'longitude'):
-                raise
-            logger.warning(
-                'Coordinates for sea ice velocity components differ. '
-                'Changing y component latitude and longitude to match the '
-                'coordinates of x component')
-            siv.remove_coord('latitude')
-            siv.remove_coord('longitude')
-
-            if isinstance(siu.coord('latitude'), DimCoord):
-                siv.add_dim_coord(siu.coord('latitude'),
-                                  siu.coord_dims('latitude'))
-                siv.add_dim_coord(siu.coord('longitude'),
-                                  siu.coord_dims('longitude'))
-            else:
-                siv.add_aux_coord(siu.coord('latitude'),
-                                  siu.coord_dims('latitude'))
-                siv.add_aux_coord(siu.coord('longitude'),
-                                  siu.coord_dims('longitude'))
+            logger.debug('Regridding siv into siu grid to compute sispeed')
+            siv = regrid(siv, siu, 'linear')
```

Comment:
is `siu` the right grid? why not regrid `siu` on `siv` grid? Or it doesn't actually matter at all

Reply:
It does not matter. In NEMO, for example, siu and siv are computed on the u and v points of the cell:

![image](https://user-images.githubusercontent.com/13482714/74156110-fb28ca00-4c15-11ea-8be3-f79739cbb2c1.png)

Any of those will be correct (in our case, probably even more correct will be to interpolate to the T point, which is usually the one used for arecello and other grid points)

Reply:
love it - a diagram!